libevm/VM.h

```diff
@@ -184,7 +185,69 @@ class VM: public VMFace
 		uint64_t w = uint64_t(v);
 		return w;
 	}
-	};
+	
+	template<class T> uint64_t toInt15(T v)
+	{
+		// check for overflow
+		if (v > 0x7FFF)
+			throwOutOfGas();
+		uint64_t w = uint64_t(v);
+		return w;
+	}
+	
+	//
+	// implementations of simd opcodes
+	//
+	// input bytes are the inline simd type descriptors for the operand vectors on the stack
+	//
+#if EIP_616
+
+	void xadd    (uint8_t);
+	void xmul    (uint8_t);
+	void xsub    (uint8_t);
+	void xdiv    (uint8_t);
+	void xsdiv   (uint8_t);
+	void xmod    (uint8_t);
+	void xsmod   (uint8_t);
+	void xlt     (uint8_t);
+	void xslt    (uint8_t);
+	void xgt     (uint8_t);
+	void xsgt    (uint8_t);
+	void xeq     (uint8_t);
+	void xzero   (uint8_t);
+	void xand    (uint8_t);
+	void xoor    (uint8_t);
+	void xxor    (uint8_t);
+	void xnot    (uint8_t);
+	void xshr    (uint8_t);
+	void xsar    (uint8_t);
+	void xshl    (uint8_t);
+	void xrol    (uint8_t);
+	void xror    (uint8_t);
+	void xmload  (uint8_t);
+	void xmstore (uint8_t);
+	void xsload  (uint8_t);
+	void xsstore (uint8_t);
+	void xvtowide(uint8_t);
+	void xwidetov(uint8_t);
+	void xpush   (uint8_t);
+	void xput    (uint8_t, uint8_t);
+	void xget    (uint8_t, uint8_t);
+	void xswizzle(uint8_t);
+	void xshuffle(uint8_t);
+	
+	u256 vtow(uint8_t _b, const u256& _in);
+	void wtov(uint8_t _b, u256 _in, u256& _o_out);
```

Comment:
CHeck comments on implementation.  Arguments and returns for this and vtow() are as they need to be/